storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileWriter.java

```diff
@@ -34,7 +35,7 @@ public class LocalFileWriter implements Closeable {
   public LocalFileWriter(File file) throws IOException {
     fileOutputStream = new FileOutputStream(file, true);
     // init fsDataOutputStream
-    dataOutputStream = new DataOutputStream(fileOutputStream);
+    dataOutputStream = new DataOutputStream(new BufferedOutputStream(fileOutputStream));
```

Comment:
Will BufferedOutputStream influence the performance of writing data file? Will it increase one time data copy when we use BufferedOutputStream?

Reply:
From the flamegraph, i dont found this problem. But for the rule of minor change, i think we should make this change only for write index. What do u think?

Reply:
I'm a little worried that the data copy will increase gc time.

Reply:
cc @colinmjj

Reply:
Compare with shuffle data, the size of index data can be ignored. And gc shouldn't be a problem with this improvement.

Reply:
@colinmjj You're right
BufferedOutpuStream's implement is as below
```
    public synchronized void write(byte b[], int off, int len) throws IOException {
        if (len >= buf.length) {
            /* If the request length exceeds the size of the output buffer,
               flush the output buffer and then write the data directly.
               In this way buffered streams will cascade harmlessly. */
            flushBuffer();
            out.write(b, off, len);
            return;
        }
        if (len > buf.length - count) {
            flushBuffer();
        }
        System.arraycopy(b, off, buf, count, len);
        count += len;
    }
```
When data is big enough, we won't copy data to buffer.
+1 for this PR